boot/zephyr/include/sysflash/sysflash.h

```diff
@@ -16,20 +16,18 @@
 
 #elif (MCUBOOT_IMAGE_NUMBER == 2)
 
-extern uint32_t _image_1_primary_slot_id[];
-
-#define FLASH_AREA_IMAGE_PRIMARY(x)            \
-        ((x == 0) ?                            \
-           PM_MCUBOOT_PRIMARY_ID :             \
-         (x == 1) ?                            \
-          (uint32_t)_image_1_primary_slot_id : \
+#define FLASH_AREA_IMAGE_PRIMARY(x)          \
+        ((x == 0) ?                          \
+           PM_MCUBOOT_PRIMARY_ID :           \
+         (x == 1) ?                          \
+           PM_MCUBOOT_PRIMARY_1_ID :         \
```

Comment:
Doesn't this break existing solution for dual primary single secondary which is used when updating mcuboot?

Reply:
Thought that it wasn't used since I didn't find any reference to it before now `nrf/subsys/bootloader/image/multi_image_mcuboot.conf` I think I forgot to test it properly.

My suggestion is to accept the limitation that you cannot use it with secure boot and #elif this with `CONFIG_SECURE_BOOT`.

Reply:
Yes I agree, we only need to support a limited use case for this, and not working with `SECURE_BOOT` is not a problem IMO.